web-client/src/view/BloodCenter/index.tsx

```diff
@@ -1,7 +1,234 @@
-import React from 'react';
-
+import React, { useState, useEffect } from 'react';
+import { useQuery } from '@apollo/client';
+import { GET_BLOOD_CENTERS } from 'API/bloodDonationCenters';
+import { IBloodDonationCenterConnection, /*IPageInfo,*/ IBloodDonationCenterEdge, IBloodDonationCenter } from 'interfaces/bloodDonationCenters';
+// import Pagination from '@mui/material/Pagination';
+import Button from 'components/Button';
+import Box from '@mui/material/Box';
+import FlexBox from 'components/StyledComponents/FlexBox';
+import Text from 'components/StyledComponents/Text';
+// import CircularProgress from '@mui/material/CircularProgress';
+import SearchInput from 'components/SearchInput';
+// import IconButton from '@mui/material/IconButton';
+// import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+// import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+// import PaginationContainer from 'components/StyledComponents/PaginationContainer';
+// import Button from '@mui/material/Button';
+// import Menu from '@mui/material/Menu';
+// import MenuItem from '@mui/material/MenuItem';
+import FilterListIcon from '@mui/icons-material/FilterList';
+import Dialog from '@mui/material/Dialog';
+import DialogContent from '@mui/material/DialogContent';
+// import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
+import Slide, { SlideProps } from '@mui/material/Slide';
+// import { TransitionProps } from '@mui/material/transitions';
+import { styled } from '@mui/material/styles';
+const Transition = React.forwardRef<HTMLDivElement, SlideProps>(
+  function Transition(props, ref) {
+    return <Slide direction="up" ref={ref} {...props} />;
+  },
+);
+const CustomDialog = styled(Dialog)(()=> ({
+  '& .MuiDialog-paper': {
+    width: '100%',
+    height: '250px',
+    borderRadius: '16px 16px 0 0',
+    backgroundColor: 'rgba(255, 255, 255, 0.9)',
+    position: 'fixed',
+    bottom: 0,
+    left: 0,
+    margin: 0,
+    maxWidth: 'none',
+    boxShadow: 'none',
+  },
+  '& .MuiBackdrop-root': {
+    backgroundColor: 'rgba(255, 255, 255, 0)', // Зробили фон прозорим
+  },
+}));
 const BloodCenters: React.FC = () => {
-  return <h1>Центри переливання крові</h1>;
+  // const { loading, error, data } = useQuery(GET_BLOOD_CENTERS);
+  // const { loading, error, data, fetchMore } = useQuery(GET_BLOOD_CENTERS, { variables: { first: 5 } });
+  // const [pageInfo, setPageInfo] = useState<IPageInfo>();
+  const [bloodCenters, setBloodCenters] = useState<IBloodDonationCenter[]>([]);
+  // const [currentPage, setCurrentPage] = useState(1);
+  const [isMobile, setIsMobile] = useState(false);
```

Comment:
Maybe we can put this to an own custom hook? So we don't need to repeat.

Reply:
I wrote about it above, I implemented it with mobx store